### PR TITLE
Track menu: add star rating

### LINKED ISSUE #10652: Change rating for more than one track at once in library view
Reported by: **[marcel-vd-berg](https://launchpad.net/~marcel-vd-berg)**
Date: 2022-01-30T11:18:36Z
Status: Confirmed
Importance: Wishlist
Launchpad Issue: [lp1959510](https://bugs.launchpad.net/bugs/1959510)
Tags: gui, library, metadata, usability

---

I mentioned this bug/wish earlier in 2016 ( https://mixxx.discourse.group/t/rating-more-than-one-track-at-once/15367 ).
Mixxx developer Daschuer suggested to fill a bug to make sure it is noticed by the Mixxx team.

I think the library rating system for Mixxx 2.3.1 Linux x86_64 is buggy and lacks functionality.
This is not about rating songs/tracks in the mp3 tag, but storing it in the database along with BPM and key.

I imported 5 iTunes playlists with corresponding star rating. Switching to these playlist give me the opportunity to mass rate the tracks.
Unfortunately the rating can only be done by single track one by one!
And while rating them one by one, often the selected rating jumps back to zero rating!

It would be great to add a library menu item "Rating" to the current list "Adjust BPM", "Select Color", "Metadata" and "Reset" in the context menu of the library (Remarkable to see the option to mass reset the rating).

I hope some developer will pickup this request to improve the rating system in Mixxx.

### PR DESCRIPTION
**Pull request by ronso0**
One of the coffee break projects to keep the balance between chore and fun/challenges.
All building blocks are already there, and working on top of #12648 this was easy. Styling was a no-brainer unexpectedly : )

image

Works as desired. Draft until #12648 is merged.
based on #12648 
Fixes #10652

_Note: I did intentionally not define a hover effect. It seems that can only be applied to the star rating (not the entire row), and since the rating is surrounded by submenu actions that expand on hover, the star hover effect kicks in only after those submenus disappeared and the pointer is moved (unlike with other actions/submenus where the hover style is applied instantly)._

### CODE REVIEW

File: src/widget/wtrackmenu.cpp [hunk 1]
```diff
@@ -151,6 +153,11 @@ void WTrackMenu::createMenus() {
         m_pBPMMenu->setTitle(tr("Adjust BPM"));
     }
 
+    if (featureIsEnabled(Feature::Properties)) {
+        m_pRatingMenu = new QMenu(this);
+        m_pRatingMenu->setTitle(tr("Set Rating"));
```
Comment (ronso0) [inline src/widget/wtrackmenu.cpp:158]: I'm not sure why the title is set in a separate step, not in the initializer.
So I simply copied how it's done everywhere else.

**Comment by ronso0** [comment]
Oh wait, why the submenu?
This could as well be inline.

**Comment by ronso0** [comment]
This can be added in a 2.4 point release or in 2.5, I don't mind.

**Comment by m0dB** [comment]
LGTM!

**Pull request merged**